app/service/app_svc.py

```diff
@@ -152,8 +152,11 @@ async def register_contacts(self):
 
     """ PRIVATE """
 
-    async def _save_configurations(self):
-        for cfg in ['default', 'agents', 'payloads']:
+    async def _save_configurations(self, main_config='default'):
+        with open('conf/%s.yml' % main_config, 'w') as config:
+            config.write(yaml.dump(self.get_config(name='main')))
+
```

Comment:
this looks like these lines can be combined back
[main_config, 'agents', 'payloads']

Reply:
yep.  It was slightly trickier, because the changes here starts differentiating between the the name of a config file and the type of config file it is.  I think overall, that's good even it it makes things a bit more complicated (I think custom config files were supposed to be supported all along, but I think when we started updating configs on server close that we broke some of that since the `conf/default.yml` path was hardcoded in).

Anyways, for loop change here: https://github.com/mitre/caldera/pull/1544/commits/49c9ebfcee0d20d910eae94639f24f5106fa1909